libyul/AST.h

```diff
@@ -35,56 +35,81 @@ namespace solidity::yul
 
 using Type = YulString;
 
-struct TypedName { langutil::SourceLocation location; YulString name; Type type; };
+struct DebugData
+{
+	explicit DebugData(langutil::SourceLocation _location): location(std::move(_location)) {}
+	langutil::SourceLocation location;
+};
+
+inline std::shared_ptr<DebugData const> createDebugData(langutil::SourceLocation _location = {})
```

Comment:
How about putting it inside `DebugData` and making it a factory function? I.e. `DebugData::create()`?

Reply:
Sounds good, and probably will look nicer. I'll update like this